### ISSUE TITLE
Rewrite the Dataset field choices functionality

### ISSUE THREAD

**Issue by vanlummelhuizen** [Signbank/Global-signbank#630]
See also #626.

Right now the form for _excluding_ field choices for a dataset, includes all datasets and all field choices in one form. Also, when updating these settings, it starts with the assumption that everything is excluded:

https://github.com/Signbank/Global-signbank/blob/4b77b1b0acfd9fb67c7a766188d96b7a8d7811c0/signbank/dictionary/update.py#L2036-L2039

When a form POST is not correct, all field choices end up excluded for all datasets.

I suggest we do 2 things:

1. Make a form per dataset
2. Only send field choices that need to be _excluded_ instead of field choices that need to be included.

**Comment by ocrasborn** [comment]
Action 2 sounds like a good idea, but I have doubts about action 1: there's value for us to see multiple datasets side by side, there'll be more people in the future who manage multiple data sets.

**Comment by vanlummelhuizen** [comment]
@ocrasborn it is possible to have multiple forms on one page so that a user can still see multiple datasets side by side.

**Comment by ocrasborn** [comment]
Ah good!

**Comment by Woseseltops** [comment]
Okay as far as I'm concerned. Just for my understanding, the main goal of suggestion 2 is to decrease the impact of future bugs?

**Comment by vanlummelhuizen** [comment]
@Woseseltops Kind of. My assessment of the code given above is that if the POST data is not as expected all field choices may be marked as excluded which is not what we want.

**Comment by susanodd** [comment]
I'm working on the issue to put frequencies #625 next to the field choices.
Yes, this page is very confusing! It looks like to exclude, you un-check.
[The template has the code 
`if choice not in excluded_choices checked="checked"`
so checked things are included, not excluded.}
(For NGT everything is checked on my local copy at first displaying the page. So nothing is excluded.)
You're right. This is a mess to look at.

**Comment by susanodd** [comment]
Probably some panels for each different field so they can be collapsed per field.

**Comment by susanodd** [comment]
Is it correct that only the Dataset Manager of a particular dataset and possibly @ocrasborn should be able to use this functionality?
(Recall that users have change_dataset permission when they are allowed to create glosses in a dataset.)

I'm logged in as user_dataset on my local machine. (I'm not in group Dataset Managers on my machine. So I view the Manage Datasets page as user_dataset. The user_dataset can only manage NGT.)

The url for this template (view) is to a function in adminviews, not a class. This seems to make the permissions work differently.

**Comment by susanodd** [comment]
I revised this in issue #625. Point 2 is completed in that issue.
Now there are separate toggle panels for each field choice category so that the list can be shortened by only showing those categories you want.

The old logic (old code) was very weird because the "default" for the Dataset objects  was to exclude all field choices. Some old code was implemented to invert this. (I rewrote the code. The original code seemed like a mockup.)

The page now shows checked boxes when a field is excluded.
(I added text to the header to state this.)
Code-wise, I created a DatasetFieldChoiceView class to accommodate additional functionality in the template.

Currently, all managed datasets are still shown on the same page, as before.
With test user-dataset (of group Dataset Manager), this user only can change NGT.
(I developed the new code testing it out as user-dataset.)
